doc/source/user/tutorial-ma.rst

```diff
@@ -0,0 +1,364 @@
+  applications
+
+**What are masked arrays?**
+
+Consider the following problem. You have a dataset with missing or invalid
+entries. If you're doing any kind of processing on this data, and want to
+`skip` or flag these unwanted entries without just deleting them, you may have
+to use conditionals or filter your data somehow. The :mod:`numpy.ma` module
+provides some of the same funcionality of
+:class:`NumPy ndarrays <numpy.ndarray>` with added structure to ensure
+invalid entries are not used in computation.
+
+From the :mod:`Reference Guide <numpy.ma>`:
+
+    A masked array is the combination of a standard :class:`numpy.ndarray` and
+    a **mask**. A mask is either ``nomask``, indicating that no value of the
+    associated array is invalid, or an array of booleans that determines for
+    each element of the associated array whether the value is valid or not.
+    When an element of the mask is ``False``, the corresponding element of the
+    associated array is valid and is said to be unmasked. When an element of
+    the mask is ``True``, the corresponding element of the associated array is
+    said to be masked (invalid).
+    
+
+We can think of a :class:`MaskedArray <numpy.ma.MaskedArray>` as a
+combination of:
+
+- Data, as a regular :class:`numpy.ndarray` of any shape or datatype;
+- A boolean mask with the same shape as the data;
+- A ``fill_value``, a value that may be used to replace the invalid entries
+  in order to return a standard :class:`numpy.ndarray`.
+
+**When can they be useful?**
+
+There are a few situations where masked arrays can be more useful than just
+eliminating the invalid entries of an array:
+
+- When you want to preserve the values you masked for later processing, without
+  copying the array;
+- When you have to handle many arrays, each with their own mask. If the mask is
+  part of the array, you avoid bugs and the code is possibly more compact;
+- If you can't avoid or eliminate missing values, but don't want to deal with
+  :class:`NaN <numpy.nan>` (Not A Number) values in your operations.
+
+Masked arrays are also a good idea since the :mod:`numpy.ma` module also
+comes with a specific implementation of most :term:`NumPy universal functions
+(ufuncs) <ufunc>`, which means that you can still apply fast vectorized
+functions and operations on masked data. The output is then a masked array.
+We'll see some examples of how this works in practice below.
+
+**Using masked arrays to see COVID-19 data**
+
+From `Kaggle <https://www.kaggle.com/atilamadai/covid19>`_ it is possible to
+download a dataset with initial data about the COVID-19 outbreak in the
+beginning of 2020. We are going to look at a small subset of this data,
+contained in the file ``who_covid_19_sit_rep_time_series.csv``.
+
+.. code::
+   
+    >>> import numpy as np
+    >>> filename = "who_covid_19_sit_rep_time_series.csv"
+
+Let's explore the data inside this file for the first 14 days of records. To
```

Comment:
I think it might be helpful here to provide a description in words of what is in the file and how it is organized.

This is aimed at researchers and people not very experienced with Python, so understanding the problem and the data will be very helpful so that they can follow the reasoning for choosing, and understanding, the programming techniques that follow.

Perhaps something like

> The data file contains data of different types and is organized as follows.
>
> The first row is a header line that (mostly) describes the data in each column that follow in the rows below, and beginning in the fourth column, the header is the date of the observation.
>
> The second through seventh row contain summary data that is of a different type than that which we are going to examine, so we will need to exclude that from the data with which we will work.
>
> The numerical data we wish to work with begins at column 4, row 8, and extends from there to the rightmost column and the lowermost row.

would be relatively short but add to fuller understanding of why you are doing what follows?

Reply:
Good point, thanks for the suggestion! I'll update it.